timemachine/fe/mcgregor.py

```diff
@@ -172,12 +185,15 @@ def mcs(
     marcs = _initialize_marcs_given_predicate(g_a, g_b, predicate)
 
     priority_idxs = tuple(tuple(x) for x in priority_idxs)
-    start_time = time.time()
+    # Keep start time for debugging purposes below
+    start_time = time.time()  # noqa
 
     # run in reverse by guessing max # of edges to avoid getting stuck in minima.
     max_threshold = _arcs_left(marcs)
     for idx in range(max_threshold):
         cur_threshold = max_threshold - idx
+        if cur_threshold < min_threshold:
```

Comment:
This is useful in the case where you have some idea of a lower bound on the number of atoms needed to map to be well converged.